include/ddc/discrete_space.hpp

```diff
@@ -194,6 +194,12 @@ std::enable_if_t<2 <= sizeof...(Args), std::tuple<Args...>> init_discrete_space(
     return detail::extract_after(std::move(a), std::index_sequence_for<Args...>());
 }
 
+/**
+ * @tparam DDim a discrete dimension
+ * @return the discrete space instance associated with `DDim`.
+ * It is recommended to call this function from a `KOKKOS_FUNCTION`.
+ * Prefer `ddc::host_discrete_space` for a host-only function.
```

Comment:
This function must be called from a `KOKKOS_FUNCTION`.
Call `ddc::host_discrete_space` for a host-only function instead.

Reply:
Yes you are right. The case we encountered is not easy to explain in one sentence. Typically if one wants to retrieved variables from a discrete dimension before a kernel, that is meant to be captured by the lambda, the user "must" call `host_discrete_space`.